packages/web/components/stepper/index.tsx

```diff
@@ -0,0 +1,323 @@
+        "flex items-center justify-center gap-5",
+        className
+      )}
+    >
+      {Array.from({ length: totalSteps }).map((_, index) => (
+        <button
+          key={index}
+          onClick={() => {
+            setActiveStep(index);
+          }}
+          className={classNames(
+            "mx-1 h-2 w-2 rounded-full",
+            index === activeStep ? "bg-osmoverse-500" : "bg-osmoverse-700"
+          )}
+        >
+          <span className="sr-only">{`Step ${index + 1}`}</span>
+        </button>
+      ))}
+    </div>
+  );
+};
+
+export const StepperRightChevronNavigation: FunctionComponent<{
+  className?: string;
+}> = ({ className }) => {
+  const { nextStep, activeStep, totalSteps } = useStepperContext();
+  return (
+    <IconButton
+      aria-label="Next step"
+      mode="unstyled"
+      icon={<Icon id="chevron-right" />}
+      className={classNames(
+        "h-3 w-3 text-osmoverse-400 disabled:cursor-not-allowed disabled:opacity-60",
+        className
+      )}
+      onClick={() => {
+        nextStep();
+      }}
+      disabled={activeStep === totalSteps - 1}
+    />
+  );
+};
+
+export const StepperLeftChevronNavigation: FunctionComponent<{
+  className?: string;
+}> = ({ className }) => {
+  const { previousStep, activeStep } = useStepperContext();
+  return (
+    <IconButton
+      aria-label="Previous step"
+      mode="unstyled"
+      icon={<Icon id="chevron-left" />}
+      className={classNames(
+        "h-3 w-3 text-osmoverse-400 disabled:cursor-not-allowed disabled:opacity-60",
+        className
+      )}
+      onClick={() => {
+        previousStep();
+      }}
+      disabled={activeStep === 0}
+    />
+  );
+};
+
+/**
+ * The Stepper component is a UI element that can be used to display content sequentially.
+ * It is an ideal choice for constructing features like slideshows, reels, or wizards where
+ * step-by-step navigation is required.
+ *
+ * It includes an autoplay feature, which allows automatic progression to the
+ * next step after a specified delay, providing a smooth, hands-free navigation experience.
+ *
+ * The Stepper is a headless component and requires multiple children to work:
+ *  - the `Step` serves to identify and render individual steps within the Stepper.
+ *    Each instance of Step corresponds to one stage in the sequential navigation.
+ *  - the `StepsIndicator` is responsible for generating a series of navigation buttons.
+ *    These buttons allow users to manually move between steps, granting a high level of navigational control.
+ *
+ * @example
+ * <Stepper autoplay={{ delayInMs: 4000, stopOnHover: true }}>
+ *  <Step>
+ *   <h4>Heading</h4>
+ *   <div className="h-12 w-12 bg-osmoverse-500"></div>
+ *   <p>test 1</p>
+ *  </Step>
+ *  <Step>
+ *   <p>test 2</p>
+ *  </Step>
+ *  <StepsIndicator />
+ * </Stepper>
+ */
+const Stepper: FunctionComponent<StepsProps> = (props) => {
+  const { children, autoplay } = props;
+
+  const timerTimeInMs = useRef(0);
```

Comment:
Why use `useRef` instead of `useState` here? Don't you want a re-render to occur when the value changes?

Reply:
@JoseRFelix can you address this please.

Reply:
We don't need to. This value is within the interval to maintain progress even when the timer is paused or stopped. Adding a `useState` would cause an excess of re-renders